diff --git a/ports/arrow/portfile.cmake b/ports/arrow/portfile.cmake
index d72442f..9384b8e 100644
--- a/ports/arrow/portfile.cmake
+++ b/ports/arrow/portfile.cmake
@@ -43,6 +43,7 @@ string(COMPARE EQUAL ${VCPKG_LIBRARY_LINKAGE} "dynamic" ARROW_BUILD_SHARED)
 string(COMPARE EQUAL ${VCPKG_LIBRARY_LINKAGE} "static" ARROW_BUILD_STATIC)
 string(COMPARE EQUAL ${VCPKG_LIBRARY_LINKAGE} "dynamic" ARROW_DEPENDENCY_USE_SHARED)
 
+z_vcpkg_setup_pkgconfig_path(BASE_DIRS "DEBUG")
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}/cpp"
     OPTIONS
diff --git a/scripts/cmake/z_vcpkg_setup_pkgconfig_path.cmake b/scripts/cmake/z_vcpkg_setup_pkgconfig_path.cmake
index b1321c0..cc1baf0 100644
--- a/scripts/cmake/z_vcpkg_setup_pkgconfig_path.cmake
+++ b/scripts/cmake/z_vcpkg_setup_pkgconfig_path.cmake
@@ -1,5 +1,5 @@
-function(z_vcpkg_setup_pkgconfig_path)
-    cmake_parse_arguments(PARSE_ARGV 0 "arg" "" "CONFIG" "")
+function(z_vcpkg_setup_pkgconfig_path arg_xxx arg_CONFIG)
+    # cmake_parse_arguments(PARSE_ARGV 0 "arg" "" "CONFIG" "")
 
     if("${arg_CONFIG}" STREQUAL "")
         message(FATAL_ERROR "CONFIG is required.")
@@ -30,10 +30,10 @@ function(z_vcpkg_setup_pkgconfig_path)
 
     foreach(prefix IN ITEMS "${CURRENT_INSTALLED_DIR}" "${CURRENT_PACKAGES_DIR}")
         vcpkg_host_path_list(PREPEND ENV{PKG_CONFIG_PATH} "${prefix}/share/pkgconfig")
-        if(arg_CONFIG STREQUAL "RELEASE")
+        if(${arg_CONFIG} STREQUAL "RELEASE")
             vcpkg_host_path_list(PREPEND ENV{PKG_CONFIG_PATH} "${prefix}/lib/pkgconfig")
             # search order is lib, share, external
-        elseif(arg_CONFIG STREQUAL "DEBUG")
+        elseif(${arg_CONFIG} STREQUAL "DEBUG")
             vcpkg_host_path_list(PREPEND ENV{PKG_CONFIG_PATH} "${prefix}/debug/lib/pkgconfig")
             # search order is debug/lib, share, external
         else()
